.github/workflows/prepare_release.yml

```diff
@@ -0,0 +1,67 @@
+---
+
+name: Prepare Release
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: 'Module version to be released.'
+        required: true
+        type: string
+      allowed_owner:
+        description: 'Only allow this owner'
+        required: true
+        type: string
+      working-directory:
+        description: The working directory where all jobs should be executed. Used for modules in subdirectories like a monorepo or a control repository.
+        default: '.'
+        required: false
+        type: string
+    secrets:
+      username:
+        description: 'The forge username to use'
+        required: true
+
+env:
+  BUNDLE_WITHOUT: development:test:system_tests
+
+jobs:
+  prepare_release:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    name: 'Puppet Forge'
+    runs-on: ubuntu-24.04
+    if: github.repository_owner == inputs.allowed_owner
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.3'
+          bundler-cache: true
+          working-directory: ${{ inputs.working-directory }}
+      - name: Update metadata.json to new version
+        run: bundle exec rake module:bump_to_version[${{ inputs.version }}]
+      - name: Update CHANGELOG.md and update REFERENCE.md if required
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          CHANGELOG_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: bundle exec rake release:prepare
+      - name: 'Check if a release is necessary'
+        id: 'check'
+        run: |
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
```

Comment:
Does this work? I think if you bump to a new version even if there aren't changes then it still creates an empty header for the new version.

Reply:
I tested it manually in [puppet-dns](https://github.com/theforeman/puppet-dns) and it appears to work:
```console
$ git log 11.0.1..HEAD
$ be rake module:bump_to_version[11.0.2]
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
Bumping version to 11.0.2
Bumping version to 11.0.2
$ changelog
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
Found 41 tags                                                          
Fetching tags dates: 41/41
Sorting tags...
Received issues: 263                                                   
Pull Request count: 239                                                
Filtered pull requests: 124
Filtered issues: 12
Fetching events for issues and PR: 136                                 
Fetching closed dates for issues: 136/136
Associating PRs with tags: 124/124
Generating entry...
Done!
Generated log placed in /home/ekohl/dev/puppet-dns/CHANGELOG.md
$ git diff CHANGELOG.md
$ echo $?
0
```

Reply:
ah great!